mogc_partnerships/views.py

```diff
@@ -170,33 +175,7 @@ def get_serializer(self, *args, **kwargs):
     def create_collection(self, validated_data, cohort):
         member_emails = [od["email"] for od in validated_data]
 
-        User = get_user_model()
-        membership_accounts = User.objects.filter(
-            email__in=[email for email in member_emails]
-        )
-        account_email_map = {user.email: user for user in membership_accounts}
-
-        cohort_memberships = [
-            CohortMembership(
-                user=account_email_map.get(member_email),
-                cohort=cohort,
-                email=member_email,
-            )
-            for member_email in member_emails
-        ]
-
-        objects = CohortMembership.objects.bulk_create(
-            cohort_memberships, ignore_conflicts=True
-        )
-        # bulk_create doesn't return autoincremented IDs with MySQL DBs
-        # so we have to query results separately
-        cohort_memberships = CohortMembership.objects.filter(
-            email__in=[cm.email for cm in objects], cohort=cohort
-        )
-
-        tasks.trigger_send_cohort_membership_invites(cohort_memberships)
```

Comment:
Hmm, I'd forgotten that the email sending was already in a task. I'm surprised that the other work being done here was timing out. I wonder if we could reproduce the timeout locally, and then see if something like [`batch_size`](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#bulk-create) makes a difference.

Reply:
I'm a little torn because I figured we'd need to make this a background take eventually, just didn't think it would be so slow for a relatively small batch.

Is the concern the lack of meaningful user feedback?

Now that we have ace working, it would be pretty easy to send an email to the admin when the job is completed with results. How's that sound?

Reply:
Ah, gotcha - I'd bet this is a bottleneck:
```
        membership_accounts = User.objects.filter(
            email__in=[email for email in member_emails]
        )
```
Do we have anything in OpenShift that would help identify the source for sure? I could log some timers otherwise.